### ISSUE TITLE
[0.10.x]Qitmeer freezing at stopping server when the ip config is invalid

### ISSUE THREAD

**Issue by forchain** [Qitmeer/qitmeer#467]
This issue only applies to 0.10.x, both cases work fine in 0.9.x

### build
0.10.0+dev-040005b

###  invalid  RPC IP
```
rpclisten=localhost:28131
```

Qitmeer displays error and tries to stop, yet failed and freezing
```
2021-04-06|19:17:28.761 [INFO ] Starting Rebroadcast                module=P2P
2021-04-06|19:17:28.761 [INFO ] try stop bm                         module=node
2021-04-06|19:17:28.761 [INFO ] Block manager shutting down         module=blkmanager
2021-04-06|19:17:28.761 [INFO ] Wait For Block manager stop ...     module=blkmanager
2021-04-06|19:17:28.761 [INFO ] Block manager stopped               module=blkmanager
2021-04-06|19:17:28.761 [INFO ] Stopping tx manager                 module=txmanager
2021-04-06|19:17:28.761 [INFO ] try stop cpu miner                  module=node
2021-04-06|19:17:28.761 [INFO ] P2P Service Stop                    module=P2P
2021-04-06|19:17:28.761 [INFO ] Rebroadcast shutting down           module=P2P
2021-04-06|19:17:28.761 [INFO ] Close BlockManager Event Subscribe  module=blkmanager
2021-04-06|19:17:28.761 [INFO ] P2P PeerSync Stop                   module=SYNC
2021-04-06|19:17:28.761 [ERROR] Uable to start server               error="'localhost' is not a valid IP address"
2021-04-06|19:17:28.761 [INFO ] Gracefully shutting down the server...
2021-04-06|19:17:28.761 [INFO ] Stopping Server                     module=node

```

Note:
Ctrl+C is not working

```
2021-04-06|19:17:28.761 [ERROR] Uable to start server               error="'localhost' is not a valid IP address"
2021-04-06|19:17:28.761 [INFO ] Gracefully shutting down the server...
2021-04-06|19:17:28.761 [INFO ] Stopping Server                     module=node
^C2021-04-06|19:21:05.031 [INFO ] Shutting down...                    Received signal=interrupt
^C2021-04-06|19:21:05.101 [INFO ] Already shutting down...            Received signal=interrupt
^C2021-04-06|19:21:05.557 [INFO ] Already shutting down...            Received signal=interrupt
^C2021-04-06|19:21:05.700 [INFO ] Already shutting down...            Received signal=interrupt
^C2021-04-06|19:21:06.083 [INFO ] Already shutting down...            Received signal=interrupt

```

**Comment by forchain** [comment]
follow up,

case 1 is not due to invalid address, it is probably due  to a loopback address
so the following config could reproduce case 1 as well

```
 listen=127.0.0.1:28130
```

**Comment by dindinw** [comment]
fixed by #468
